src/pages/Search/index.js

```diff
@@ -3,17 +3,19 @@ import { useParams } from 'react-router-dom';
 import loadData from '../../api/posts';
 import Loader from '../../components/loader';
 import Form from './components/Form';
+import Heatmap from './components/Heatmap';
 import SearchContainer from './styles';
 
 const SearchPage = () => {
   const [loading, setLoading] = useState(false);
+  const [posts, setPosts] = useState({});
   const { subreddit } = useParams();
-
   useEffect(() => {
     setLoading(true);
     const onLoad = async () => {
       try {
-        await loadData(subreddit);
+        const res = await loadData(subreddit);
+        setPosts(res);
       } catch (error) {
         console.log(error);
```

Comment:
You could set an error here to an additional state, like `setError('An error occured`)` and render it below